src/urllib3/_collections.py

```diff
@@ -45,54 +100,67 @@ class RecentlyUsedContainer(MutableMapping):
         ``dispose_func(value)`` is called.  Callback which will get called
     """
 
-    ContainerCls = OrderedDict
+    _container: "OrderedDict[_KT, _VT]"
+    _maxsize: int
+    dispose_func: Optional[Callable[[_VT], None]]
+    lock: RLock
 
-    def __init__(self, maxsize=10, dispose_func=None):
+    def __init__(
+        self, maxsize: int = 10, dispose_func: Optional[Callable[[_VT], None]] = None
+    ) -> None:
+        super().__init__()
         self._maxsize = maxsize
         self.dispose_func = dispose_func
-
-        self._container = self.ContainerCls()
+        self._container = OrderedDict()
```

Comment:
Adding to the breaking changes list "You can't set `RecentlyUsedContainer.ContainerCls` anymore". Totally fine with this change, this class is internal and shouldn't be mucked with.